addon/components/mdc-drawer.js

```diff
@@ -0,0 +1,123 @@
+  onopen: x => x,
+  onclose: x => x,
+  //endregion
+
+  //region ComputedProperties
+  isPermanent: computed('permanent', 'persistent', 'temporary', function () {
+    const {permanent, persistent, temporary} = getProperties(this, 'permanent', 'persistent', 'temporary');
+    if (permanent && (temporary || persistent)) {
+      throw new Ember.Error('Cannot be permanent and temporary or persistent');
+    }
+    return permanent;
+  }),
+  isPersistent: computed('permanent', 'persistent', 'temporary', function () {
+    const {permanent, persistent, temporary} = getProperties(this, 'permanent', 'persistent', 'temporary');
+    if (persistent && (permanent || temporary)) {
+      throw new Ember.Error('Cannot be persistent and permanent or temporary');
+    }
+    return persistent;
+  }),
+  isTemporary: computed('permanent', 'persistent', 'temporary', function () {
+    const {permanent, persistent, temporary} = getProperties(this, 'permanent', 'persistent', 'temporary');
+    if (temporary && (permanent || persistent)) {
+      throw new Ember.Error('Cannot be temporary and permanent or persistent');
+    }
+    return temporary;
+  }),
+  //endregion
+
+  //region Methods
+  updateOpenness() {
+    const foundation = get(this, 'foundation');
+    if (!foundation) { return; }
+    const open = get(this, 'open');
+    if (foundation.isOpen() && !open) {
+      foundation.close();
+    }
+    if (!foundation.isOpen() && open) {
+      foundation.open();
+    }
+  },
+  /**
+   * @returns {MDCPersistentDrawerFoundation|MDCTemporaryDrawerFoundation}
+   */
+  createFoundation() {
+    const {permanent, persistent} = getProperties(this, 'permanent', 'persistent');
+    if (permanent) {
+      return { init() {}, destroy() {} };
+    }
+    const Foundation = persistent ? MDCPersistentDrawerFoundation : MDCTemporaryDrawerFoundation;
+    const {FOCUSABLE_ELEMENTS, ITEMS_SELECTOR, DRAWER_SELECTOR} = Foundation.strings;
+    const adapter = {
+      addClass: className => get(this, 'mdcClasses').addObject(className),
+      removeClass: className => get(this, 'mdcClasses').removeObject(className),
+      hasClass: className => get(this, 'element.classList').contains(className),
+      registerInteractionHandler: (type, handler) => this.registerMdcInteractionHandler(type, handler),
+      deregisterInteractionHandler: (type, handler) => this.deregisterMdcInteractionHandler(type, handler),
+      registerDrawerInteractionHandler: (type, handler) => this.registerMdcInteractionHandler(type, handler),
+      deregisterDrawerInteractionHandler: (type, handler) => this.deregisterMdcInteractionHandler(type, handler),
+      registerTransitionEndHandler: (handler) => this.registerMdcInteractionHandler('transitionend', handler),
+      deregisterTransitionEndHandler: (handler) => this.deregisterMdcInteractionHandler('transitionend', handler),
+      registerDocumentKeydownHandler: (handler) => this.registerMdcInteractionHandler('keydown', handler),
```

Comment:
I believe the `registerDocument` (and de-register) handlers shouldn't use `registerMdcInteractionHandler` (or de-register) because MDC-Web wants them attached to the `document` element, and that method attaches them to the component's element.

Reply:
The [docs](https://github.com/material-components/material-components-web/tree/master/packages/mdc-drawer#using-the-foundation-class) just say that it attaches to the _"root"_ element, not the document.

So I think that using these methods are fine.

Reply:
:+1: OK cool. Google's gotta get better at naming things.

Reply:
Maybe they updated it since you looked, but the docs say attach to the document object.

![image](https://user-images.githubusercontent.com/552093/31895902-080128be-b7d8-11e7-9515-044c5a77494b.png)